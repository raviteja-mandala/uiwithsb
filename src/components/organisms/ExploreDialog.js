import { Box, Button, Grid, makeStyles } from "@material-ui/core";
import PropTypes from "prop-types";
import React, { useEffect, useState } from "react";
import { Link } from "react-router-dom";
import CategoryWithIcon from "../molecules/CategoryWithIcon";

const useStyles = makeStyles(() => ({
  dialogContent: {
    width: "100%",
    marginTop: 0,
    marginLeft: 0,
    marginRight: 0,
    height: 100,
    display: "block",
  },
  blinkistbody: {
    flex: 1,
    display: "flex",
    flexDirection: "column",
  },

  categoryName: {
    "&:hover": {
      color: "blue",
      backgroundColor: "#f2f2f2",
    },
    color: "black",
    paddingTop: 10,
    paddingBottom: 10,
  },

  categoryLink: {
    textDecoration: "none",
  },

  dialogContainer: {
    width: "100%",
    marginTop: 0,
    marginLeft: 0,
    marginRight: 0,
    height: 400,
    backgroundColor: "#f2f2f2",
  },

  exploreTable: {
    position: "fixed",
    left: 0,
    right: 0,
    backgroundColor: "#f2f2f2",
    display: "flex",
    width: "100%",
    flexDirection: "column",
    top: 60,
    zIndex: 2,
  },

  hideTable: {
    display: "none",
  },

  tableContainer: {
    marginLeft: 300,
    marginTop: 30,
    marginRight: 300,
  },

  exploreHeaderTitles: {
    paddingTop: 5,
    paddingBottom: 5,
    border: "none",
    "&:hover": {
      color: "blue",
      backgroundColor: "#f2f2f2",
    },

    headerTitlesContainer: {
      borderBottom: 2,
      marginBottom: 5,
    },

    categoriesContainer: {
      marginTop: 5,
      marginBottom: 10,
    },
  },
}));

function ExploreDialog(props) {
  const classes = useStyles();
  const [categories, setCategories] = useState(() => {
    return [];
  });
  const allBooks = useState(() => {
    return [];
  });
  const refresh = props.open;
  const categoriesArray = [];
  const allBooksArray = [];
  const handleDialogClose = () => {
    props.onclose(false);
  };

  const fetchCategories = () => {
    fetch("http://localhost:3000/books")
      .then((response) => response.json())
      .then((data) => {
        console.log(data);
        data.map((book) => {
          categoriesArray.push(book.category);
          allBooksArray.push(book);
        });
        setCategories([...new Set(categoriesArray)]);
        allBooks[1]([...allBooksArray]);
      });
  };

  useEffect(() => {
    fetchCategories();
  }, [refresh]);

  return (
    <Box
      className={props.open === true ? classes.exploreTable : classes.hideTable}
    >
      <Box className={classes.tableContainer}>
        <Grid container direction="column">
          <Box borderBottom={0.05}>
            {" "}
            <Grid
              item
              container
              direction="row"
              className={classes.headerTitlesContainer}
            >
              <Grid item xs={3}>
                <Button className={classes.exploreHeaderTitles}>
                  Explore by category
                </Button>
              </Grid>
              <Grid item xs={3}>
                <Button className={classes.exploreHeaderTitles}>
                  See recently added titles
                </Button>
              </Grid>
              <Grid item xs={3}>
                <Button className={classes.exploreHeaderTitles}>
                  See popular titles
                </Button>
              </Grid>
            </Grid>
          </Box>

          <Grid item container className={classes.categoriesContainer}>
            {categories.map((category) => {
              return (
                <Grid item xs={4}>
                  <Link
                    to={{ pathname: `/explore/${category}` }}
                    className={classes.categoryLink}
                    onClick={handleDialogClose}
                  >
                    <Box className={classes.categoryName}>
                      <CategoryWithIcon message={category}></CategoryWithIcon>
                    </Box>
                  </Link>
                </Grid>
              );
            })}
          </Grid>
        </Grid>
      </Box>
    </Box>
  );
}

ExploreDialog.propTypes = {
  open: PropTypes.bool,
  onclose: PropTypes.func
};

export default ExploreDialog;
