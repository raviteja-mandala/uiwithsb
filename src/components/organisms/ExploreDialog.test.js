import {
    render,
    screen
} from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import ExploreDialog from './ExploreDialog';

// beforeEach(() => {
//     jest.spyOn(global, 'fetch').mockResolvedValue({
//         json: jest.fn().mockResolvedValue(books)
//     })
// });

test('test explore dialog ', async () => {

    const books = [{

        "id": 6,
        "bookTitle": "Zoology",
        "bookAuthor": "Vikram",
        "min": 20,
        "reads": "0.7k",
        "image": "book_image1.png",
        "category": "Medicine"

    }];

    const mockFetchMethod = jest.spyOn(global, 'fetch').mockImplementation(() => Promise.resolve({ json: () => Promise.resolve(books) }));
    render(<BrowserRouter><ExploreDialog open={true} onclose={() => { }} /></BrowserRouter>);
    const myElement = await screen.findByText(/Medicine/i);
    expect(myElement).toBeInTheDocument();
    mockFetchMethod.mockRestore;
});


test('test explore dialog 1', async () => {

    const books = [
        {
            "id": 3,
            "bookTitle": "Fluid Mechanics",
            "bookAuthor": "Rohan",
            "min": 42,
            "reads": "1.2k",
            "image": "book_image3.png",
            "category": "Science"
        },
    ];
    const mockFetchMethod = jest.spyOn(global, 'fetch').mockImplementation(() => Promise.resolve({ json: () => Promise.resolve(books) }));
    render(<BrowserRouter><ExploreDialog open={true} onclose={() => { }} /></BrowserRouter>);
    const myElement = await screen.findByText(/Science/i);
    expect(myElement).toBeInTheDocument();
    mockFetchMethod.mockRestore;

    const tree = renderer.create((<BrowserRouter><ExploreDialog open={true} onclose={() => { }} /></BrowserRouter>)).toJSON();
    expect(tree).toMatchSnapshot();
});