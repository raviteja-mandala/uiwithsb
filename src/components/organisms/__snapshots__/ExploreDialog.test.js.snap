// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test explore dialog 1 1`] = `
<div
  className="MuiBox-root MuiBox-root-31 makeStyles-exploreTable-21"
>
  <div
    className="MuiBox-root MuiBox-root-32 makeStyles-tableContainer-23"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
    >
      <div
        className="MuiBox-root MuiBox-root-33"
      >
         
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-item"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-exploreHeaderTitles-24"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Explore by category
              </span>
            </button>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-exploreHeaderTitles-24"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                See recently added titles
              </span>
            </button>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-exploreHeaderTitles-24"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                See popular titles
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item"
      />
    </div>
  </div>
</div>
`;
